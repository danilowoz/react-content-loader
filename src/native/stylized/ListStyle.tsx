import * as React from 'react'

import Holder, { Rect } from '../Holder'
import { IContentLoaderProps } from '../'

const ReactContentLoaderListStyle: React.FC<IContentLoaderProps> = props => (
  <Holder {...props}>
    <Rect x="0" y="0" rx="3" ry="3" width="250" height="10" />
    <Rect x="20" y="20" rx="3" ry="3" width="220" height="10" />
    <Rect x="20" y="40" rx="3" ry="3" width="170" height="10" />
    <Rect x="0" y="60" rx="3" ry="3" width="250" height="10" />
    <Rect x="20" y="80" rx="3" ry="3" width="200" height="10" />
    <Rect x="20" y="100" rx="3" ry="3" width="80" height="10" />
  </Holder>
)

export default ReactContentLoaderListStyle
