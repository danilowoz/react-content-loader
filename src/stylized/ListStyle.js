import React from 'react'
import Wrap from '../Wrap'

const ListStyle = (props) => {
    return (
        <Wrap {...props}>
            <rect x="0" y="0" rx="3" ry="3" width="250" height="10" />
            <rect x="20" y="20" rx="3" ry="3" width="220" height="10" />
            <rect x="20" y="40" rx="3" ry="3" width="170" height="10" />
            <rect x="0" y="60" rx="3" ry="3" width="250" height="10" />
            <rect x="20" y="80" rx="3" ry="3" width="200" height="10" />
            <rect x="20" y="100" rx="3" ry="3" width="80" height="10" />
        </Wrap>
    )
}

export default ListStyle
