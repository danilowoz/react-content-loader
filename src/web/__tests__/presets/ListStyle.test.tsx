import * as React from 'react'
import * as renderer from 'react-test-renderer'

import ListStyle from '../../presets/ListStyle'

describe('ListStyle', () => {
  const wrapper = renderer.create(
    <ListStyle uniqueKey="ListStyle" speed={20} />
  )

  test('renders correctly', () => {
    const tree = wrapper.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('props are propagated ', () => {
    expect(wrapper.root.props.speed).toEqual(20)
    expect(wrapper.root.props.uniqueKey).toEqual('ListStyle')
  })
})
