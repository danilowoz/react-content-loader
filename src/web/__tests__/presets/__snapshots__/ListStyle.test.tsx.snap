// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListStyle renders correctly 1`] = `
<svg
  aria-labelledby="ListStyle-aria"
  id={null}
  role="img"
  style={Object {}}
  viewBox="0 0 400 110"
>
  <title
    id="ListStyle-aria"
  >
    Loading...
  </title>
  <rect
    clipPath="url(#ListStyle-diff)"
    height="100%"
    role="presentation"
    style={
      Object {
        "fill": "url(#ListStyle-animated-diff)",
      }
    }
    width="100%"
    x="0"
    y="0"
  />
  <defs
    role="presentation"
  >
    <clipPath
      id="ListStyle-diff"
    >
      <rect
        height="10"
        rx="3"
        ry="3"
        width="250"
        x="0"
        y="0"
      />
      <rect
        height="10"
        rx="3"
        ry="3"
        width="220"
        x="20"
        y="20"
      />
      <rect
        height="10"
        rx="3"
        ry="3"
        width="170"
        x="20"
        y="40"
      />
      <rect
        height="10"
        rx="3"
        ry="3"
        width="250"
        x="0"
        y="60"
      />
      <rect
        height="10"
        rx="3"
        ry="3"
        width="200"
        x="20"
        y="80"
      />
      <rect
        height="10"
        rx="3"
        ry="3"
        width="80"
        x="20"
        y="100"
      />
    </clipPath>
    <linearGradient
      id="ListStyle-animated-diff"
    >
      <stop
        offset="0%"
        stopColor="#f5f6f7"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="20s"
          keyTimes="0; 0.25; 1"
          repeatCount="indefinite"
          values="-2; -2; 1"
        />
      </stop>
      <stop
        offset="50%"
        stopColor="#eee"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="20s"
          keyTimes="0; 0.25; 1"
          repeatCount="indefinite"
          values="-1; -1; 2"
        />
      </stop>
      <stop
        offset="100%"
        stopColor="#f5f6f7"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="20s"
          keyTimes="0; 0.25; 1"
          repeatCount="indefinite"
          values="0; 0; 3"
        />
      </stop>
    </linearGradient>
  </defs>
</svg>
`;
